sqld/src/http/either.rs

```diff
@@ -0,0 +1,168 @@
+// Code borrowed from https://docs.rs/http-body-util/0.1.0-rc.3/http_body_util/enum.Either.html and
+// we don't use some of the methods.
+#![allow(unused)]
+
+use std::error::Error;
+use std::fmt::Debug;
+use std::pin::Pin;
+use std::task::{Context, Poll};
+
+use bytes::Buf;
+use http_body::{Body, SizeHint};
+use proj::EitherProj;
+
+/// Sum type with two cases: [`Left`] and [`Right`], used if a body can be one of
+/// two distinct types.
+///
+/// [`Left`]: Either::Left
+/// [`Right`]: Either::Right
+#[derive(Debug, Clone, Copy)]
+pub enum Either<L, R> {
```

Comment:
we already have transitive dependency to either, might as well bring in the crate

Reply:
@MarinPostma pretty sure it was needed to be able to `impl Body for Either` below? But I really hope we can do without it if we migrate everything to axum :crossed_fingers:

Reply:
ah yeah, orphan rule...

Reply:
This is a special one since it needs the body impl that I don't think the other one has?